### ISSUE TITLE
[flash_ctrl/otbn/lc_ctrl] Consider routing the RMA signals back to LC in parallel

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#19136]
### Description

After a discussion with @andreaskurth and @vogelpi we dertermined that it may be better to route the RMA ack signals from `otbn` and `flash_ctrl` back to `lc_ctrl` in parallel due to security considerations. We went with the daisy-chain approach in #13499 since it required fewer RTL modifications at the time.

We should consider fixing this for PROD.

CC @cdgori @johannheyszl @moidx

**Comment by vogelpi** [comment]
Thanks for creating the issue @msfschaffner , let's discuss this in the next Sec WG meeting.

For context, there are two things that need to be considered around CDCs and FI vulnerabilities of the RMA request signal.

1. We demand the following for `lc_ctrl` signals
   > For all life cycle signals except the escalation signal ESCALATE_EN, all values different from ON must be interpreted as OFF in RTL. In case of ESCALATE_EN, all values different from OFF must be interpreted as ON in RTL.

   This means checking for illegal encodings and escalating upon detection isn't something that is foreseen. When daisy-chaining the RMA req/ack signals this opens up an attack window: After LC initiates a secure wipe of `flash_ctrl` and `otbn` and when waiting for `flash_ctrl` to be done (takes a long time) an attacker could try to glitch the RMA req signal going into `otbn`. If the attacker manages to flip all bits at the same time or one after another `otbn` would then perform a secure wipe (very quick) and signal the ack to `lc_ctrl` . `lc_ctrl` would then reboot into RMA without the `flash_ctrl` finishing the secure wipe first. That's why we've added the detection for illegal encodings inside OTBN (see #19050).

2. Daisy-chaining with detection of illegal encodings of the RMA req signal works fine for Discrete only because `flash_ctrl` is in the same clock domain as `otbn`. This means `otbn` will always see a valid encoding for the RMA request signal. If an invalid encoding is observed this is indeed a strong sign that something fishy is going on. Other top-levels might require a CDC on that path, i.e., there will be staggering effects on the RMA req when switching from one to another state resulting in illegal combinations and `otbn` always aborting RMA entry.

The cleanest way of solving this is to:
1. Remove the check for illegal encodings on the RMA req signal inside `otbn` to make it comply with the `lc_ctrl` specification. See #19050.
2. Route RMA req/ack signals to relevant blocks in parallel and perform spurious ack detection inside `lc_ctrl` to avoid the FI vulnerability.

**Comment by msfschaffner** [comment]
Thanks @vogelpi for summarizing this. Indeed, I agree with this proposal. 2. is possible, but it will need an interface change in the life cycle controller. Spurious ack detection is however already implemented in the current FSM and can  be extended for a second input.

**Comment by vogelpi** [comment]
We've discussed this in the Sec WG meeting yesterday and concluded to move forward with the proposal outlined above. I am thus removing the hotlist label.

**Comment by GregAC** [comment]
It's worth considering this for PROD, as whilst it might not be an actual issue it gives a cleaner implementation of a security feature. Lower priority though.
